src/engraving/libmscore/tempotext.cpp

```diff
@@ -473,6 +603,88 @@ void TempoText::layout()
     autoplaceSegmentElement();
 }
 
+//---------------------------------------------------------
+//   parseEquation
+//---------------------------------------------------------
+
+void TempoText::parseEquation()
+{
+    static const QRegularExpression equationExpression("[\\[\\(]?[\uECA0-\uECB7]+ *= *[\uECA0-\uECB70-9]+[\\]\\)]?");
+
+    static const std::map<QString, QString> equationMap = {
+        { "d", "\uECA0" },
+        { "w", "\uECA2" },
+        { "h", "\uECA3" },
+        { "q", "\uECA5" },
+        { "e", "\uECA7" },
+        { "s", "\uECA9" },
+        { "t", "\uECAB" },
+        { ".", "\uECB7" }
+    };
+
+    if (_isEquationVisible) {
+        QString equation = _equation;
+
+        for (auto replacement : equationMap) {
+            equation.replace(replacement.first, replacement.second);
+        }
+
+        if (equationExpression.match(xmlText()).hasMatch()) {
+            setXmlText(xmlText().replace(equationExpression, equation));
+        } else {
+            setXmlText(xmlText() + equation);
+        }
+    } else {
+        setXmlText(xmlText().replace(equationExpression, ""));
+    }
+
+    if (_followText) {
+        updateTempo();
+    }
+}
+
+//---------------------------------------------------------
+//   isEquationValid
+//---------------------------------------------------------
+
+bool TempoText::isEquationValid(const QString equation) const
+{
+    const QRegularExpression equationExpression("^[\\[,\\(]?[ ]*([dwhqest].*) *= *(\\d+|[dwhqest].*)[\\],\\)]?$");
+    QRegularExpressionMatch match = equationExpression.match(equation);
+
+    return match.hasMatch();
+}
+
+//---------------------------------------------------------
+//   getRelativeDuration
+//---------------------------------------------------------
+
+float TempoText::getRelativeDuration(const QString marking) const
+{
+    static const std::map<QString, float> durationMap = {
+        { "d", 8.0f },
+        { "w", 4.0f },
+        { "h", 2.0f },
+        { "q", 1.0f },
+        { "e", 0.5f },
+        { "s", 0.25f },
+        { "t", 0.125f },
+    };
+
+    float baseDuration = 1.0f;
+
+    for (auto duration : durationMap) {
+        if (marking[0] == duration.first) {
+            baseDuration = duration.second;
+            break;
+        }
+    }
+
+    const int dots = marking.count(".");
+
+    return baseDuration * std::powf(1.5, dots);
```

Comment:
or cast it to float here, in order to get it to cimpile on Linux, like in GitHib's CI

Reply:
`std::pow(3, dots)/std::pow(2, dots)` would arguably be more efficient